### PR TITLE
Add tests to fetch and render all docs in Validation

### PR DESCRIPTION
**Pull request by timothyfroehlich**
This uses the test fixtures added in the base CL and generates tests for all entries in Validation's examples list, skipping any that don't use a Figma Doc (one of them doesn't).

TO TEST:

We do not have a way to run Instrumentation tests or Figma fetches via GitHub actions. The tests added here must be performed manually. 

To watch and/or debug the tests: Launch an emulator and run `./gradlew connectedTest`

To run the tests automatically using automated test devices: run `./gradlew tabletAtdApi30DebugAndroidTest`

To run the full test suite: run `./dev-scripts/test-all.sh`

### CODE REVIEW

File: integration-tests/validation/src/androidTest/java/com/android/designcompose/testapp/validation/FetchAndRenderExamples.kt [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        @Parameterized.Parameters(name = "{0}")
+        fun data(): List<Array<Any>> {
+            return EXAMPLES.filter { it.third != null }
+                .map { arrayOf(it.first, it.second, it.third!!) }
+        }
+    }
+
+    @Test
+    fun testFetchAndRender() {
+        composeTestRule.setContent(fileComposable)
```
Comment (rylin8) [inline integration-tests/validation/src/androidTest/java/com/android/designcompose/testapp/validation/FetchAndRenderExamples.kt:55]: How are `fileComposable` and `fileId` populated?

Comment (timothyfroehlich) [reply integration-tests/validation/src/androidTest/java/com/android/designcompose/testapp/validation/FetchAndRenderExamples.kt:55]: Via the companion object which sets up the parameterized tests:
https://www.rossharper.net/2016/02/parameterized-junit4-unit-tests-in-kotlin/

**Comment by timothyfroehlich** [comment]
Meant to set this back to draft, I need to update it based on the review comments from the base PR.

File: integration-tests/validation/src/androidTest/java/com/android/designcompose/testapp/validation/figmaIntegrationTests/FetchAndRenderExamples.kt [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+         *
+         * Convert Validation's examples list of triples into a list of arrays, so that they can be
+         * used as parameters for the tests
+         *
+         * @return List of test parameters for the Parameterized runner
+         */
+        @JvmStatic
+        @Parameterized.Parameters(name = "{0}")
+        fun initializeTestData(): List<Array<Any>> =
+            EXAMPLES.filter { it.third != null }.map { arrayOf(it.first, it.second, it.third!!) }
```
Comment (rylin8) [inline integration-tests/validation/src/androidTest/java/com/android/designcompose/testapp/validation/figmaIntegrationTests/FetchAndRenderExamples.kt:76]: How is this function called and how are the member variables of this class populated?

Comment (timothyfroehlich) [reply integration-tests/validation/src/androidTest/java/com/android/designcompose/testapp/validation/figmaIntegrationTests/FetchAndRenderExamples.kt:76]: Sorry, I swore I'd sent a reply to this before. It's the normal way of doing [Junit4 Parameterized](https://www.tutorialspoint.com/junit/junit_parameterized_test.htm) tests, but with the correct annotations for Kotlin. Here's some other examples: https://gist.github.com/rossharper/8f6c3c169b6b5c23e12c

File: integration-tests/validation/src/main/java/com/android/designcompose/testapp/validation/MainActivity.kt [hunk 3]
```diff
@@ -114,40 +114,45 @@ val interFont =
...
+        Triple(
+            "Grid Layout Documentation",
+            { GridLayoutDocumentation() },
+            "MBNjjSbzzKeN7nBjVoewsl"
+        ),
+        Triple("Blend Modes", { BlendModeTest() }, "ZqX5i5g6inv9tANIwMMXUV"),
+        Triple("Vector Rendering", { VectorRenderingTest() }, "Z3ucY0wMAbIwZIa6mLEWIK"),
+        Triple("Dials Gauges", { DialsGaugesTest() }, "lZj6E9GtIQQE4HNLpzgETw"),
+        Triple("Masks", { MaskTest() }, "mEmdUVEIjvBBbV0kELPy37"),
+        Triple("Variable Borders", { VariableBorderTest() }, "MWnVAfW3FupV4VMLNR1m67")
```
Comment (rylin8) [inline integration-tests/validation/src/main/java/com/android/designcompose/testapp/validation/MainActivity.kt:155]: Would be nice if we didn't have to have the doc ID here and in the test itself, but that can be looked into later.

**Review by rylin8 — APPROVED** [review]

File: integration-tests/validation/src/main/java/com/android/designcompose/testapp/validation/MainActivity.kt (hunk 3 above)
Comment (timothyfroehlich) [reply integration-tests/validation/src/main/java/com/android/designcompose/testapp/validation/MainActivity.kt:155]: Yeah, one thing we could do is make it available via the generated doc (Like `HelloWorldDoc.fileId`. Can look into it other ways too.

**Pull request merged**